haskell/private/coverage_wrapper.sh.tpl

```diff
@@ -42,9 +43,15 @@ $binary_path "$@"
 $hpc_path report $tix_file_path $hpc_dir_args > __hpc_coverage_report
 echo "Overall report"
 cat __hpc_coverage_report
-expression_coverage=$(grep "expressions used" __hpc_coverage_report | cut -c 1-3)
-if [ $expression_coverage -lt $expected_expression_coverage ]
+covered_expression_percentage=$(grep "expressions used" __hpc_coverage_report | cut -c 1-3)
+if [ $covered_expression_percentage -lt $expected_covered_expressions_percentage ]
 then
-  echo -e "\n==>$ERRORCOLOR Inadequate expression coverage.$CLEARCOLOR Expected $expected_expression_coverage%, but actual coverage was $ERRORCOLOR$(($expression_coverage))%$CLEARCOLOR.\n"
+  echo -e "\n==>$ERRORCOLOR Inadequate expression coverage percentage.$CLEARCOLOR Expected $expected_covered_expressions_percentage%, but actual coverage was $ERRORCOLOR$(($covered_expression_percentage))%$CLEARCOLOR.\n"
+  exit 1
+fi
+uncovered_expression_count=$(grep "expressions used" __hpc_coverage_report | sed s/.*\(//g | cut -f1 -d "/")
+if [ $uncovered_expression_count -gt $expected_uncovered_expression_count ]
```

Comment:
Could you implement a ratcheting mechanism? If you only check greater than then noone will tighten the bounds. What about having both `expected_uncovered_expression_count` and `max_expected_uncovered_expression_count` and then doing something like:
```
# check max allowed
if [ $uncovered_expression_count -gt $max_expected_uncovered_expression_count ]
then
  echo "Too many uncovered expressions..."
  exit 1
fi

# check exact number
if [ $uncovered_expression_count -ne $expected_uncovered_expression_count ]
then
  echo "You have xxx uncovered expressions but expected YYY."

  if [ $uncovered_expression_count -lt $expected_uncovered_expression_count ]
  then
    echo "Nice job! You got rid of uncovered expressions, please reduce the expected number..."
  elif [ $uncovered_expression_count -gt $expected_uncovered_expression_count ]
    echo "Uh oh please don't make test coverage worse...."
  fi

  exit 1
fi
```